cmd/lotus-shed/balances.go

```diff
@@ -941,3 +949,335 @@ var fillBalancesCmd = &cli.Command{
 		return nil
 	},
 }
+
+var chainFip0081PledgeCmd = &cli.Command{
+	Name:        "fip0081-pledge",
+	Description: "Calculate sector pledge values comparing current to pre-FIP-0081",
+	ArgsUsage:   "[epoch number]",
+	Action: func(cctx *cli.Context) error {
+
+		ctx := lcli.ReqContext(cctx)
+
+		api, acloser, err := lcli.GetFullNodeAPIV1(cctx)
+		if err != nil {
+			return err
+		}
+		defer acloser()
+
+		var ts *types.TipSet
+		if cctx.Args().Present() {
+			epoch, err := strconv.ParseInt(cctx.Args().First(), 10, 64)
+			if err != nil {
+				return xerrors.Errorf("parsing epoch arg: %w", err)
+			}
+			ts, err = api.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(epoch), types.EmptyTSK)
+			if err != nil {
+				return err
+			}
+		} else {
+			ts, err = api.ChainHead(ctx)
+			if err != nil {
+				return err
+			}
+		}
+
+		cases := []struct {
+			sectorSize   abi.SectorSize
+			verifiedSize uint64
+			duration     abi.ChainEpoch
```

Comment:
I was surprised to learn that duration had no apparent impact on pledge .. does that seem right @ZenGround0 ?

Reply:
[Yes this sounds right](https://github.com/filecoin-project/builtin-actors/blob/master/actors/miner/src/monies.rs#L254).  The rough idea with pledge is there are two terms.  The first is ~20 days of expected rewards the second contributes to locking up a fixed target of the circulating supply  and the sector contributes to this target proportional to its power.